algo/gtree/tree_test.go

```diff
@@ -73,58 +73,51 @@ func TestTree_FindNotFound(t *testing.T) {
 	require.Empty(t, value)
 }
 
-var sizes = []int{1e6}
-
-//func BenchmarkInsert(b *testing.B) {
-//	for _, n := range sizes {
-//		b.Run(fmt.Sprintf("BenchmarkInsert_%d", n), func(b *testing.B) {
-//			s := generateRandomSlice(n)
-//			b.ResetTimer()
-//
-//			for i := 0; i < b.N; i++ {
-//				getTree(s)
-//			}
-//		})
-//	}
-//}
+var sizes = []int{1e2, 1e3, 1e4, 1e6}
 
-func BenchmarkFind(b *testing.B) {
+func BenchmarkInsert(b *testing.B) {
 	for _, n := range sizes {
-		b.Run(fmt.Sprintf("BenchmarkFind_%d", n), func(b *testing.B) {
-			s := generateRandomSlice(n)
-			b.Log("after generateRandomSlice")
-
+		b.Run(fmt.Sprintf("BenchmarkInsert_%d", n), func(b *testing.B) {
+			s := helper.GenerateRandomSliceSet(n)
+			key := helper.Higher(s) + 1
 			tr := getTree(s)
-			b.Log("after getree")
-
-			key := randRange(0, n)
-			b.Log("after rand")
 
 			b.ResetTimer()
-			b.Log("after timer")
 			for i := 0; i < b.N; i++ {
-				tr.Find(key)
+				tr.Insert(key, key)
```

Comment:
The problem I see with this approach is that we are reusing the same `tr` instance `b.N` times:
- Every iteration mutates the `tr` instance adding new elements
- The `key` variable is not updated with the latest `tr` state

My point here is that `key` would be only the worst case in the first iteration, but not in the rest. 

Maybe I am missing something here?

Anyway, for having an estimation it was more than fair scenario.

Reply:
Yes i think you are totally right :+1:  , i couldn't find better solution than this :sweat: Maybe we can take a look at this again on the next iteration :)